### PR TITLE
improves error handling

### PR DESCRIPTION
**Pull request by OhDavit**
improves error handling and uses request in a way that stubbing request.function is possible.

### CODE REVIEW

**Comment by OhDavit** [comment]
can someone please review?

File: lib/sabre-dev-studio.js [hunk 1]
```diff
@@ -100,18 +99,18 @@ var SabreDevStudio = (function() {
...
             log.error('Error:', error);
             cb(error, data, response);
-          } else if (!response) {
-            var err = Error.http(response.statusCode, errorCodes[response.statusCode]);
-            cb(err, data, response);
           } else if (response.statusCode === 200) {
             cb(null, data, response);
           } else if (response.statusCode === 401 && retryCount >= 0) {
             fetchAccessToken(endpoint, options, cb, --retryCount);
+          } else if (errorCodes[response.statusCode]) {
```
Comment (barrettclark) [inline lib/sabre-dev-studio.js]: This is slick

File: lib/sabre-dev-studio.js [hunk 2]
```diff
@@ -100,18 +99,18 @@ var SabreDevStudio = (function() {
             "bearer": this.config.access_token
           }
         };
-        request(requestUrl, reqestOptions, function(error, response, data) {
+        request[method.toLowerCase()](requestUrl, reqestOptions, function(error, response, data) {
```
Comment (barrettclark) [inline lib/sabre-dev-studio.js]: When I originally wrote this library I used deprecated functionality in the OAuth library. That was a couple of years ago. This is good functionality, and is nice to have. My only question is whether a deeper rewrite to modernize is called for, and if that is something you would be interested in looking doing?

**Comment by barrettclark** [comment]
:shipit:

**Comment by barrettclark** [comment]
Thank you for submitting this. I'm sorry it took me a couple of days to look at it. It's been a crazy week.

File: lib/sabre-dev-studio.js (hunk 2 above)
Comment (OhDavit) [inline lib/sabre-dev-studio.js]: The implementation still looks nice and I don't think there is a need to rewrite the code.
Next step in my opinion should be upgrading or supporting also the latest version of sabre api: 1.9.5.1.

**Comment by OhDavit** [comment]
@barrettclark no worries and thank you for your time :)

**Pull request merged**